Modules/_sqlite/cache.c

```diff
@@ -253,6 +258,20 @@ PyObject* pysqlite_cache_display(pysqlite_Cache* self, PyObject* args)
     Py_RETURN_NONE;
 }
 
+static PyType_Slot pysqlite_NodeType_slots[] = {
+    {Py_tp_dealloc, pysqlite_node_dealloc},
+    {Py_tp_new, PyType_GenericNew},
+    {0, NULL},
+};
+
+static PyType_Spec pysqlite_NodeType_spec = {
+    .name = MODULE_NAME ".Node",
+    .basicsize = sizeof(pysqlite_Node),
+    .flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE | Py_TPFLAGS_HEAPTYPE,
```

Comment:
@vstinner 
Is it safe to port heap type if the type has `Py_TPFLAGS_BASETYPE` flag?
ref: https://github.com/python/cpython/pull/20974#discussion_r442797635